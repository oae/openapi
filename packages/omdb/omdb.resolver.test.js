const { request } = require('graphql-request');
const { gql } = require('@openapi/core/utils');
const { createServer, destroyServer } = require('@openapi/core/testUtils');

const { name: pluginName } = require('./package.json');

let server = null;

beforeAll(async () => {
  server = await createServer([
    [
      pluginName,
      {
        auth: {
          apiKey: process.env.OA_OMDB_API_KEY,
        },
      },
    ],
  ]);
});

afterAll(async () => {
  await destroyServer(server);
});

const movieFields = gql`
  fragment movieFields on Movie {
    title
    year
    imdb {
      id
      rating
      votes
    }
    metacritic {
      rating
    }
    rottenTomatoes {
      rating
    }
    rated
    released
    runtime
    genre
    director
    writer
    actors
    plot
    language
    country
    awards
    poster
    boxOffice
    production
    website
  }
`;

const validMovie = {
  title: expect.not.toBeEmpty(),
  year: expect.not.toBeEmpty(),
  imdb: {
    id: expect.not.toBeEmpty(),
    rating: expect.not.toBeEmpty(),
    votes: expect.not.toBeEmpty(),
  },
  rated: expect.not.toBeEmpty(),
  released: expect.not.toBeEmpty(),
  runtime: expect.not.toBeEmpty(),
  genre: expect.toBeArray(),
  director: expect.not.toBeEmpty(),
  writer: expect.not.toBeEmpty(),
  actors: expect.toBeArray(),
  plot: expect.not.toBeEmpty(),
  language: expect.toBeArray(),
  country: expect.toBeArray(),
  boxOffice: expect.not.toBeEmpty(),
  production: expect.not.toBeEmpty(),
  website: expect.not.toBeEmpty(),
};

describe('omdb', () => {
  it('should return movies with title', async () => {
    const query = gql`
      ${movieFields}
      {
        movies(title: "Inception", limit: 2) {
          ...movieFields
        }
      }
    `;

    const result = await request(server.endpoint, query);
    expect(result.movies).toBeArray();
    expect(result.movies[0]).toMatchObject(validMovie);
  });

  it('should return a movie for given imdbId', async () => {
    const query = gql`
      ${movieFields}
      {
        movie(imdbId: "tt1375666") {
          ...movieFields
        }
      }
    `;

    const result = await request(server.endpoint, query);
    expect(result.movie).toMatchObject(validMovie);
  });
});
