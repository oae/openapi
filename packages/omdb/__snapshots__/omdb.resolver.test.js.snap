// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`omdb should return a movie for given imdbId 1`] = `
Object {
  "movie": Object {
    "actors": Array [
      "Leonardo DiCaprio",
      "Joseph Gordon-Levitt",
      "Ellen Page",
      "Tom Hardy",
    ],
    "awards": "Won 4 Oscars. Another 152 wins & 204 nominations.",
    "boxOffice": "$292,568,851",
    "country": Array [
      "USA",
      "UK",
    ],
    "director": "Christopher Nolan",
    "genre": Array [
      "Action",
      "Adventure",
      "Sci-Fi",
      "Thriller",
    ],
    "imdb": Object {
      "id": "tt1375666",
      "rating": "8.8",
      "votes": "1,755,015",
    },
    "language": Array [
      "English",
      "Japanese",
      "French",
    ],
    "metacritic": Object {
      "rating": "74/100",
    },
    "plot": "A thief who steals corporate secrets through the use of dream-sharing technology is given the inverse task of planting an idea into the mind of a CEO.",
    "poster": "https://m.media-amazon.com/images/M/MV5BMjAxMzY3NjcxNF5BMl5BanBnXkFtZTcwNTI5OTM0Mw@@._V1_SX300.jpg",
    "production": "Warner Bros. Pictures",
    "rated": "PG-13",
    "released": "16 Jul 2010",
    "rottenTomatoes": Object {
      "rating": "86%",
    },
    "runtime": "148 min",
    "title": "Inception",
    "website": "http://inceptionmovie.warnerbros.com/",
    "writer": "Christopher Nolan",
    "year": "2010",
  },
}
`;

exports[`omdb should return movies with title 1`] = `
Object {
  "movies": Array [
    Object {
      "actors": Array [
        "Leonardo DiCaprio",
        "Joseph Gordon-Levitt",
        "Ellen Page",
        "Tom Hardy",
      ],
      "awards": "Won 4 Oscars. Another 152 wins & 204 nominations.",
      "boxOffice": "$292,568,851",
      "country": Array [
        "USA",
        "UK",
      ],
      "director": "Christopher Nolan",
      "genre": Array [
        "Action",
        "Adventure",
        "Sci-Fi",
        "Thriller",
      ],
      "imdb": Object {
        "id": "tt1375666",
        "rating": "8.8",
        "votes": "1,755,015",
      },
      "language": Array [
        "English",
        "Japanese",
        "French",
      ],
      "metacritic": Object {
        "rating": "74/100",
      },
      "plot": "A thief who steals corporate secrets through the use of dream-sharing technology is given the inverse task of planting an idea into the mind of a CEO.",
      "poster": "https://m.media-amazon.com/images/M/MV5BMjAxMzY3NjcxNF5BMl5BanBnXkFtZTcwNTI5OTM0Mw@@._V1_SX300.jpg",
      "production": "Warner Bros. Pictures",
      "rated": "PG-13",
      "released": "16 Jul 2010",
      "rottenTomatoes": Object {
        "rating": "86%",
      },
      "runtime": "148 min",
      "title": "Inception",
      "website": "http://inceptionmovie.warnerbros.com/",
      "writer": "Christopher Nolan",
      "year": "2010",
    },
    Object {
      "actors": Array [
        "Leonardo DiCaprio",
        "Joseph Gordon-Levitt",
        "Lukas Haas",
      ],
      "awards": "N/A",
      "boxOffice": "N/A",
      "country": Array [
        "USA",
      ],
      "director": "Ian Kirby",
      "genre": Array [
        "Animation",
        "Short",
        "Sci-Fi",
      ],
      "imdb": Object {
        "id": "tt5295894",
        "rating": "7.8",
        "votes": "1,138",
      },
      "language": Array [
        "English",
      ],
      "metacritic": null,
      "plot": "This Inception prequel unfolds courtesy of a beautiful Motion Comic, and explains how Cobb, Arthur and Nash were enlisted by Cobol Engineering. Diehard fans of the film will be especially interested in this one.",
      "poster": "https://m.media-amazon.com/images/M/MV5BMjE0NGIwM2EtZjQxZi00ZTE5LWExN2MtNDBlMjY1ZmZkYjU3XkEyXkFqcGdeQXVyNjMwNzk3Mjk@._V1_SX300.jpg",
      "production": "N/A",
      "rated": "N/A",
      "released": "07 Dec 2010",
      "rottenTomatoes": null,
      "runtime": "15 min",
      "title": "Inception: The Cobol Job",
      "website": "N/A",
      "writer": "Christopher Nolan (based on characters by)",
      "year": "2010",
    },
  ],
}
`;
